worker/src/ocall_bridge/bridge_api.rs

```diff
@@ -0,0 +1,80 @@
+/*
+    Copyright 2019 Supercomputing Systems AG
+    Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+use crate::ocall_bridge::component_factory::OCallBridgeComponentFactory;
+use lazy_static::lazy_static;
+use log::*;
+use parking_lot::RwLock;
+use sgx_types::{
+    sgx_epid_group_id_t, sgx_platform_info_t, sgx_quote_nonce_t, sgx_quote_sign_type_t,
+    sgx_report_t, sgx_spid_t, sgx_status_t, sgx_target_info_t, sgx_update_info_bit_t,
+};
+use std::sync::Arc;
+use std::vec::Vec;
+
+#[cfg(test)]
+use mockall::predicate::*;
+#[cfg(test)]
+use mockall::*;
+
+lazy_static! {
+    static ref COMPONENT_FACTORY: RwLock<Option<Arc<dyn OCallBridgeComponentFactory + Send + Sync>>> =
+        RwLock::new(None);
+}
+
+pub struct Bridge {}
+
+impl Bridge {
```

Comment:
The `Bridge` is the static/global interface to inject concrete implementations (or rather the factories for them) - this is done at startup of the worker. On the other side, it is used by the o-call FFI to retrieve the state and forward calls to their respective implementation.

Reply:
Is there any reason you're not putting such comments directly into the code? I think (atleast for me) it would be really helpful to be able to look up such comments also later on.

Reply:
Good input. This is also helpful for fellow sdk users.

Reply:
Yes, indeed, there probably should be more comments, especially in the future for all the public facing traits and functions. 

I've neglected them in part because of laziness, but also because comments tend to go out of sync with the code VERY fast, and then they're more harmful than anything else. And here we're dealing with the beginning of some substantial refactoring, where everything is very volatile.. so far my excuse. Once we reach a more stable state, we should definitely do more documentation inside the code.

Reply:
My latest commit added some of the comments you (rightfully) requested